src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc

```diff
@@ -266,49 +268,71 @@ void ResetGlobalPropertiesRequest::on_event(const event_engine::Event& event) {
-          std::max(ui_result_, tts_result_));
-    }
+  if (IsPendingResponseExist()) {
+    LOG4CXX_WARN(logger_, "unable to find application: " << connection_key());
+    return;
+  }
 
-    SendResponse(result,
-                 static_cast<mobile_apis::Result::eType>(result_code),
-                 return_info,
-                 &(message[strings::msg_params]));
+  mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
+  std::string response_info;
+  const bool result = PrepareResponseParameters(result_code, response_info);
 
-    if (!application) {
-      LOG4CXX_ERROR(logger_, "NULL pointer");
-      return;
-    }
+  SendResponse(result,
+               static_cast<mobile_apis::Result::eType>(result_code),
+               response_info.empty() ? NULL : response_info.c_str(),
+               &(message[strings::msg_params]));
+
+  if (!application) {
+    LOG4CXX_ERROR(logger_, "NULL pointer");
+    return;
+  }
+
+  if (result) {
+    application->UpdateHash();
+  }
+}
 
-    if (result) {
-      application->UpdateHash();
+bool ResetGlobalPropertiesRequest::PrepareResponseParameters(
+    mobile_apis::Result::eType& result_code, std::string& response_info) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using namespace helpers;
+
+  ResponseInfo ui_properties_info(ui_result_, HmiInterfaces::HMI_INTERFACE_UI);
+  ResponseInfo tts_properties_info(tts_result_,
+                                   HmiInterfaces::HMI_INTERFACE_TTS);
+
+  const bool result =
+      ((hmi_apis::Common_Result::SUCCESS == ui_result_) &&
+       (hmi_apis::Common_Result::SUCCESS == tts_result_ ||
+        hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == tts_result_)) ||
+      ((hmi_apis::Common_Result::SUCCESS == ui_result_) &&
+       (hmi_apis::Common_Result::INVALID_ENUM == tts_result_)) ||
+      ((hmi_apis::Common_Result::INVALID_ENUM == ui_result_) &&
+       (hmi_apis::Common_Result::SUCCESS == tts_result_));
+
+  if (result) {
+    if (hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == tts_result_) {
+      result_code = mobile_apis::Result::WARNINGS;
+      response_info =
+          std::string("Unsupported phoneme type sent in a prompt").c_str();
+    } else {
+      result_code = static_cast<mobile_apis::Result::eType>(
+          std::max(ui_result_, tts_result_));
+      response_info = MergeInfos(tts_properties_info,
+                                 tts_response_info_,
+                                 ui_properties_info,
+                                 ui_response_info_);
     }
   } else {
-    LOG4CXX_WARN(logger_, "unable to find application: " << connection_key());
+    result_code = static_cast<mobile_apis::Result::eType>(
+        std::max(ui_result_, tts_result_));
+    response_info = MergeInfos(tts_properties_info,
+                               tts_response_info_,
+                               ui_properties_info,
+                               ui_response_info_);
   }
```

Comment:
@dcherniev I saw this code snippet above. Please rework this in order to aboid duplication of code